uber/site_sections/groups.py

```diff
@@ -1,4 +1,101 @@
 from uber.common import *
+from uber.custom_tags import pluralize
+
+
+_entangled_ribbons = set(
+    getattr(c, r, -1) for r in ['BAND', 'DEPT_HEAD_RIBBON', 'PANELIST_RIBBON'])
+
+
+def _is_attendee_disentangled(attendee):
+    """
+    Returns True if the attendee has an unpaid badge and does not have any
+    other roles in the system.
+    """
+    return attendee.paid not in [c.HAS_PAID, c.NEED_NOT_PAY] \
+        and _entangled_ribbons.isdisjoint(attendee.ribbon_ints) \
+        and not attendee.admin_account \
+        and not attendee.shifts
+
+
+def _is_dealer_convertable(attendee):
+    """
+    Returns True if a waitlisted dealer can be converted into a new, unpaid
+    attendee badge.
+    """
+    return attendee.badge_type == c.ATTENDEE_BADGE \
+        and _is_attendee_disentangled(attendee)
+        # It looks like a lot of dealers have helpers that didn't get assigned
+        # a dealer ribbon. We still want to convert those badges, so we can't
+        # trust they'll have a dealer ribbon. I think this is safe because we
+        # won't even get this far if it isn't a dealer group in the first place
+        # and c.DEALER_RIBBON in attendee.ribbon_ints
+
+
+def _decline_and_convert_dealer_group(session, group, convert=False):
+    """
+    Deletes the waitlisted dealer group and converts all of the group members
+    to the appropriate badge type. Unassigned, unpaid badges will be deleted.
+    """
+    admin_note = 'Converted badge from waitlisted dealer group "{}".'.format(
+        group.name)
+
+    if not group.is_unpaid:
+        group.tables = 0
+        for attendee in group.attendees:
+            attendee.append_admin_note(admin_note)
+            attendee.ribbon = remove_opt(attendee.ribbon_ints, c.DEALER_RIBBON)
+        return 'Group dealer status removed'
```

Comment:
If the group already paid for themselves, then we strip the dealer status, and make no other changes.

Reply:
Great idea! This theoretically is impossible with our current workflows, but it's better to have a fallback in case we change anything. :+1: